.github/workflows/ci.yml

```diff
@@ -144,10 +144,95 @@ jobs:
           path: ${{ github.workspace }}/dist
           key: nginx-kubernetes-gateway-${{ github.run_id }}-${{ github.run_number }}
 
+  conformance-tests:
+    name: Gateway Conformance Tests
+    runs-on: ubuntu-22.04
+    needs: [vars, binary]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Fetch Cached Artifacts
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+        with:
+          path: ${{ github.workspace }}/dist
+          key: nginx-kubernetes-gateway-${{ github.run_id }}-${{ github.run_number }}
+
+      - name: Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+      
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175 # v4.6.0
+        with:
+          images: |
+            name=ghcr.io/nginxinc/nginx-kubernetes-gateway
+          tags: |
+            type=semver,pattern={{version}}
+            type=edge
+            type=ref,event=pr
+            type=ref,event=branch,suffix=-rc,enable=${{ startsWith(github.ref, 'refs/heads/release') }}
+
+      - name: Prepare NKG files
+        run: |
+          nkg_prefix=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 1)
+          nkg_tag=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 2)
+          yq -i "with(.spec.template.spec.containers[0]; .image = \"${nkg_prefix}:${nkg_tag}\" | .imagePullPolicy = \"Never\")" deploy/manifests/deployment.yaml
+        shell: bash
+
+      - name: Build Docker Image
```

Comment:
Would this be worth doing in our regular build steps of our pipeline? It would validate that our image builds properly, outside of the scope of conformance tests. Then we can pull in the artifact for the test run.

Reply:
GitHub makes this really awkward - long story short, we do run the build image job separately in our regular pipeline, but it's not possible to share the artifacts between the jobs unless we push them to a remote, and (generally) we don't want to push images before we've tested them (and we don't push PR images anyway). This means we have to run the build step twice, which is less than ideal, but at least we are using caching so most the layers will be reused.

Reply:
Oh you're right, we do already build the image. Not sure how I missed that.

If we save the docker file as a tarfile, would that allow us to share between jobs using the "Fetch" job?

Reply:
Yes, we could share the image between jobs in this way, but I don't think it would work for our use case. The docker build action below builds and pushes for the main branch and for the release branch - I don't think you can push to a remote using a loaded tar file using the action. We could investigate this option for PRs only though (we do need the image again to run the trivy scanner), it might save some time (or it might not - our image building is pretty quick, and uploading and downloading has it's own time penalties). I think it seems a bit out of scope for this PR though.